bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/DbLedgerStorage.java

```diff
@@ -287,23 +300,10 @@ public void initialize(ServerConfiguration conf, LedgerManager ledgerManager, Le
         ledgerIndex = new LedgerMetadataIndex(conf, KeyValueStorageRocksDB.factory, baseDir, stats);
         entryLocationIndex = new EntryLocationIndex(conf, KeyValueStorageRocksDB.factory, baseDir, stats);
 
-        // build the ledger info cache
-        int concurrencyLevel = Math.max(1, Math.max(conf.getNumAddWorkerThreads(), conf.getNumReadWorkerThreads()));
-        RemovalListener<Long, TransientLedgerInfo> ledgerInfoRemovalListener = this::handleLedgerEviction;
-        CacheBuilder<Long, TransientLedgerInfo> builder = CacheBuilder.newBuilder()
-            .initialCapacity(conf.getFileInfoCacheInitialCapacity())
-            .maximumSize(conf.getOpenFileLimit())
-            .concurrencyLevel(concurrencyLevel)
-            .removalListener(ledgerInfoRemovalListener);
-        if (conf.getFileInfoMaxIdleTime() > 0) {
-            builder.expireAfterAccess(conf.getFileInfoMaxIdleTime(), TimeUnit.SECONDS);
-        }
-        transientLedgerInfoCache = builder.build(new CacheLoader<Long, TransientLedgerInfo>() {
-            @Override
-            public TransientLedgerInfo load(Long key) throws Exception {
-                return new TransientLedgerInfo(key, ledgerIndex);
-            }
-        });
+        transientLedgerInfoCache = new ConcurrentLongHashMap<>(16 * 1024,
+                Runtime.getRuntime().availableProcessors() * 2);
```

Comment:
better to configure the concurrency level to be ` Math.max(1, Math.max(conf.getNumAddWorkerThreads(), conf.getNumReadWorkerThreads()))`?

Reply:
One thing to consider here is that if the `addWorkerThreads` is 0, it means the writes are coming directly from the io threads, and there the number depends on the number of cores.

for the `ConcurrentLongHashMap` the "parallelism" equates to the number of sections in the hash-map. Each section has its own read-write mutex. Since we only have 1 instance of this map per bookie, I don't think it's a problem to have multiple sections, since the memory increase is really tiny.